components/molecules/Table.jsx

```diff
@@ -140,26 +148,32 @@ class Table extends Component {
                     </Header>
                 )}
                 <Body>
-                    {data.map((rowData, i) => (
-                        <Row
-                            key={`row-${i}`}
-                            className={onRowClick && '-cursor--pointer'}
-                            isInverse={isInverse}
-                            isStriped={isStriped}
-                            onClick={onRowClick && (() => onRowClick(rowData))}>
-                            {columns.map((column, k) => {
-                                const isString = typeof column === 'string';
-                                const cellData = isString ? rowData[column] : rowData[column.key];
-                                const decorator = column.dataCellDecorator;
-                                const key = `row-${i}-key-${k}`;
-                                return decorator ? (
-                                    decorator(cellData, key, rowData)
-                                ) : (
-                                    <Data key={key}>{cellData}</Data>
-                                );
-                            })}
-                        </Row>
-                    ))}
+                    {data.map((rowData, i) => {
+                        const rowKey =
```

Comment:
this line states that if some rows have "customRowKey" use that as the key and otherwise use index, this will use a mix of customKeys and indexes if some of items miss customRowKey. I'm thinking we should add a custom propType validation for 'data', and make sure if customRowKey is set, check all the items in 'data' have the customRowKey set. There are examples of custom validation in https://reactjs.org/docs/typechecking-with-proptypes.html